### PR TITLE
DP-21043: Build data listing view with minimum styles

### PR DESCRIPTION
**Pull request by weekbeforenext**
**Description:**
Created Data Listing All and Data Listing Topic views with page displays, a shared template file

**Jira:** (Skip unless you are MA staff)
https://jira.mass.gov/browse/DP-21043

**To Test:**
- [ ] Go to /data-listing/all.
- [ ] Verify the Filter by Topic field only shows top level topics.
- [ ] Verify you can filter by topic and keyword.
- [ ] Go to /data-listing/topic/health.
- [ ] Verify the page title includes the topic, like "Health Data Listing".
- [ ] Verify the Filter by Sub Topic field only shows the sub topics for the given topic, "health".
- [ ] Verify you can filter by topic and keyword.

**Screenshots/GIFs:**

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: docroot/themes/custom/mass_theme/mass_theme.theme [hunk 1]
```diff
@@ -4451,6 +4492,15 @@ function mass_theme_preprocess_page(&$variables) {
...
     mass_theme_add_regions_to_node($allowed_regions, $variables);
   }
+
+  // Add a main_class for the Data Listing views pages.
+  $data_listing_view_ids = [
+    'data_listing_all',
+    'data_listing_topic',
+  ];
+  if (isset($variables['page']['content']['system_main']['#view_id']) && in_array($variables['page']['content']['system_main']['#view_id'], $data_listing_view_ids)) {
+    $variables['main_class'] = 'data-listing-placeholder';
```
Comment (weekbeforenext) [inline docroot/themes/custom/mass_theme/mass_theme.theme]: @clairesunstudio This is where the placeholder class has been added for the data listing views pages.

File: docroot/sites/development.services.yml [hunk 2]
```diff
@@ -4,6 +4,13 @@
 # 'example.settings.local.php' file, which sits next to this file.
 parameters:
   http.response.debug_cacheability_headers: true
+  twig.config:
```
Comment (clairesunstudio) [inline docroot/sites/development.services.yml]: @weekbeforenext I noticed that everytime I run `ahoy comi` these changes get overridden. I think this is generated by something else.

**Comment by clairesunstudio** [comment]
Tested on chrome, safari, firefox, IE edge, IE11 (with minor visual degradation in the teaser eyebrow)
External review environment: feature1 
https://massgovfeature1.prod.acquia-sites.com/mockup-massachusetts-data-hub---unpublished?auHash=83aHo2s6ddHVjwE13yBwZN47Mrjw7NATBWQWQtGukcs

Before merging:
- [x] need to update the breadcrumb paths in the view yml files
- [ ] need to copy and update custom code on the data hub homepage on the production site unpublished page

File: docroot/themes/custom/mass_theme/mass_theme.theme (hunk 1 above)
Comment (weekbeforenext) [reply docroot/themes/custom/mass_theme/mass_theme.theme]: @clairesunstudio Did this class need to change? Is it being used?

Comment (clairesunstudio) [reply docroot/themes/custom/mass_theme/mass_theme.theme]: This is no longer used can you remove this? Sorry about it

File: docroot/sites/development.services.yml (hunk 2 above)
Comment (weekbeforenext) [reply docroot/sites/development.services.yml]: I added a message to Teams to see how to preserve these changes for local dev. It looks like they are being replaced by core scaffold files in the composer.lock file.

Comment (weekbeforenext) [reply docroot/sites/development.services.yml]: I think this is fixed in composer now.

File: conf/drupal/config/views.view.data_listing_all.yml [hunk 3]
```diff
@@ -382,7 +382,7 @@ display:
           empty: true
           tokenize: false
           content:
-            value: '<div class="ma__breadcrumb--views"><ol><li><a href="https://edit.mass.gov/mockup-massachusetts-data-hub---unpublished?auHash=83aHo2s6ddHVjwE13yBwZN47Mrjw7NATBWQWQtGukcs">Data Home</a></li></ol></div>'
```
Comment (clairesunstudio) [inline conf/drupal/config/views.view.data_listing_all.yml]: This change breaks the breadcrumb link

File: conf/drupal/config/views.view.data_listing_topic.yml [hunk 4]
```diff
@@ -372,7 +372,7 @@ display:
           empty: false
           tokenize: false
           content:
-            value: '<div class="ma__breadcrumb--views"><ol><li><a href="https://edit.mass.gov/mockup-massachusetts-data-hub---unpublished?auHash=83aHo2s6ddHVjwE13yBwZN47Mrjw7NATBWQWQtGukcs">Data Home</a></li><li><a href="/data-listing/all">All Data Listing</a></li></ol></div>'
```
Comment (clairesunstudio) [inline conf/drupal/config/views.view.data_listing_topic.yml]: This change breaks the breadcrumb link

File: conf/drupal/config/views.view.data_listing_topic.yml [hunk 5]
```diff
@@ -372,7 +372,7 @@ display:
...
+            value: '<div class="ma__breadcrumb--views"><ol><li><a href="/node/560446">Data Home</a></li><li><a href="/data-listing/all">All Data Listing</a></li></ol></div>'
```
Comment (clairesunstudio) [inline conf/drupal/config/views.view.data_listing_topic.yml:379]:
```suggestion
            value: '<div class="ma__breadcrumb--views"><ol><li><a href="https://edit.mass.gov/mockup-massachusetts-data-hub---unpublished?auHash=83aHo2s6ddHVjwE13yBwZN47Mrjw7NATBWQWQtGukcs">Data Home</a></li><li><a href="/data-listing/all">All Data Listing</a></li></ol></div>'
```

File: conf/drupal/config/views.view.data_listing_all.yml [hunk 6]
```diff
@@ -382,7 +382,7 @@ display:
...
+            value: '<div class="ma__breadcrumb--views"><ol><li><a href="/node/560446">Data Home</a></li></ol></div>'
```
Comment (clairesunstudio) [inline conf/drupal/config/views.view.data_listing_all.yml:375]:
```suggestion
            value: '<div class="ma__breadcrumb--views"><ol><li><a href="https://edit.mass.gov/mockup-massachusetts-data-hub---unpublished?auHash=83aHo2s6ddHVjwE13yBwZN47Mrjw7NATBWQWQtGukcs">Data Home</a></li></ol></div>'
```

**Review by clairesunstudio — COMMENTED** [review]
update breadcrumb links

**Comment by clairesunstudio** [comment]
2dbeda5 and 4d0504d are for external review purposes, as the landing page is an unpublished page.
Revert 2dbeda5 and 4d0504d before merging

**Pull request merged**